### PR TITLE
improv: local bootstrap node set up w/ example

### PR DESCRIPTION
**Pull request by elielnfinic**
## Overview 
This PR enhances the Developer Experience (DX) for the local environment. It streamlines the process of running a topology bootstrap on a local device or network, making it easier to connect peers efficiently.

Currently, the changes are applied to the chat example, but if approved, I can extend this implementation to the other examples as well.

## Summary of Changes

Here are the edited files : 
- examples/chat/package.json
- examples/chat/src/index.ts
- packages/network/src/node.ts
- packages/node/configs/node.json
- README.md

Here are the created files : 
- examples/chat/src/vite-env.d.ts
- packages/node/configs/local-bootstrap.json

Feel free to let me know if there are any adjustments or suggestions. Once approved, I can proceed with applying the same improvements to the other examples.

### CODE REVIEW

File: packages/network/src/node.ts [hunk 1]
```diff
@@ -63,6 +64,26 @@ export class TopologyNetworkNode {
...
+
+		const peer_discovery = bootstrap_nodes_list.length
+			? [
+					_pubsubPeerDiscovery,
+					bootstrap({
+						list: bootstrap_nodes_list,
+					}),
+				]
+			: [_pubsubPeerDiscovery];
+
```
Comment (d-roak) [inline packages/network/src/node.ts:86]: why this change?

File: packages/network/src/node.ts [hunk 2]
```diff
@@ -102,7 +111,9 @@ export class TopologyNetworkNode {
 				}),
 				webRTC(),
 				webRTCDirect(),
-				webSockets(),
+				webSockets({
+					filter: filters.all,
+				}),
```
Comment (d-roak) [inline packages/network/src/node.ts:116]: why this change?

File: packages/node/configs/node.json [hunk 3]
```diff
@@ -1,9 +1,7 @@
 {
 	"network_config": {
 		"addresses": ["/webrtc"],
-		"bootstrap_peers": [
-			"/ip4/127.0.0.1/tcp/50000/ws/p2p/12D3KooWC6sm9iwmYbeQJCJipKTRghmABNz1wnpJANvSMabvecwJ"
-		],
+		"bootstrap_peers": [],
```
Comment (d-roak) [inline packages/node/configs/node.json:4]: why this change?

File: examples/chat/src/vite-env.d.ts
Comment (d-roak) [inline examples/chat/src/vite-env.d.ts:1]: this is adding complexity to the example

File: examples/chat/package.json [hunk 4]
```diff
@@ -10,6 +10,7 @@
     "build": "vite build",
     "clean": "rm -rf dist/ node_modules/",
     "dev": "vite serve",
+    "local": "VITE_TOPOLOGY_MODE=local vite serve",
```
Comment (d-roak) [inline examples/chat/package.json:13]: adding complexity

File: README.md [hunk 5]
```diff
@@ -52,3 +53,21 @@ pnpm install
...
+- Go to `packages/node` and run the following command to start the node:
+```bash
+pnpm cli relay --config configs/local-bootstrap.json
+```
+Keep the terminal open and copy the `peerId` from the output. You will need it to connect to the node.
+- Open a new terminal and go to `examples/chat` and run the following command to start the example:
+```bash
+pnpm local
+```
+- Make sure you update the `peerId` in the `src/index.ts` file with the one you copied from the node.
```
Comment (d-roak) [inline README.md:72]: not clear which index are you talking about here

File: README.md [hunk 6]
```diff
@@ -52,3 +53,21 @@ pnpm install
...
+- You should now be able to see the chat example running in your browser opening the displayed URL.
```
Comment (d-roak) [inline README.md:73]: the main readme is not supposed to go through the examples

File: packages/network/src/node.ts (hunk 1 above)
Comment (elielnfinic) [reply packages/network/src/node.ts:86]: If you disconnect WiFi and the `relay.droak.sh` is still in the list of bootstrap_peers, when the relay is run it produces an issue.

File: packages/network/src/node.ts (hunk 2 above)
Comment (elielnfinic) [reply packages/network/src/node.ts:116]: This allows to connect to `ws` if not, you can only connect to `wss`

File: packages/node/configs/node.json (hunk 3 above)
Comment (elielnfinic) [reply packages/node/configs/node.json:4]: This was while testing, I think, I can bring back this since I have `bootstrap-local.json`

File: examples/chat/package.json (hunk 4 above)
Comment (elielnfinic) [reply examples/chat/package.json:13]: I thought this would allow to simply run an example with `pnpm local` and connect to the local bootstrap node for [easy node configuration set up](https://github.com/elielnfinic/ts-topology/blob/5fe96b9c8d83ef19d8d144d50e9d2cd682080abd/examples/chat/src/index.ts#L32)

File: README.md (hunk 5 above)
Comment (elielnfinic) [reply README.md:72]: My bad, I should add `examples/chat/src/index.ts`

File: README.md (hunk 6 above)
Comment (elielnfinic) [reply README.md:73]: Noted, thank you.

File: packages/network/src/node.ts (hunk 1 above)
Comment (d-roak) [reply packages/network/src/node.ts:86]: what do you mean? this change is restructuring code, is not changing logic

File: packages/network/src/node.ts (hunk 2 above)
Comment (d-roak) [reply packages/network/src/node.ts:116]: we are not supposed to allow `ws`, that's right. The question is if it detects that we are working in the localhost, does the ws fail? Localhost usually has special rules (same applies for http/https)

File: examples/chat/src/vite-env.d.ts
Comment (elielnfinic) [reply examples/chat/src/vite-env.d.ts:1]: I have removed this file.

File: packages/network/src/node.ts (hunk 1 above)
Comment (elielnfinic) [reply packages/network/src/node.ts:86]: When there is no bootstrap node list, I shouldn't have the `bootstrap({list : []})` property. 
If it is in the object, I am getting the following error : 
```
@topology-foundation/node@0.2.0 cli /Users/monordi/apps/topology/ts-topology/packages/node
> tsx ./src/run.ts "relay" "--config" "configs/local-bootstrap.json"

/Users/monordi/apps/topology/ts-topology/node_modules/.pnpm/@libp2p+bootstrap@11.0.7/node_modules/@libp2p/bootstrap/src/index.ts:98
      throw new Error('Bootstrap requires a list of peer addresses')
            ^


Error: Bootstrap requires a list of peer addresses```

That's why I created the condition to check if I have bootstrap nodes in my config. If not, pass the `peer-discovery` not having the bootstrap property.

File: packages/network/src/node.ts (hunk 2 above)
Comment (elielnfinic) [reply packages/network/src/node.ts:116]: Yes, `ws` is failing in localhost

**Review by d-roak — APPROVED** [review]
lgtm
